script.js

```diff
@@ -1,4 +1,165 @@
+// === REQUIREMENTS ===
+// 1. 2 players play and they take turns
+// 2. when a player clicks submit, the game rolls 2 dice and show the output of the dice rolls e.g 2 and 4
+// 3. that player picks the order of the dice they want. e.g if he wanted 42, specify 2nd dice goes first
+// 4. after both players rolled and chosen dice order, scores are compared and higher number wins
+
+// === problem breakdown and planning ===
+// ver 1. rolls 2 dice and returns the output for 1 player. that player chooses the dice order and gets the correct output value returned
+// ver 2. refactor code to include player 2's turn
+//      - global variables for currentPlayer; allPlayersScore
+//      - refactor outputMessages to interact with each player 1 and 2
+//      - write logic for player 1 to go first then player 2, then proceed to compare score
+// ver 3. implement comparing dice scores and declare winner
+// ver 4. reset game so that players can play again without refreshing browser page
+
+// === code start ===
+// global variables
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayersScore = [];
+
+// helper function
+var rollDice = function () {
+  console.log("Control Flow: Start of rollDice()");
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  console.log("rollDice output, randomInteger: " + randomInteger);
+  return randomInteger;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control Flow: Start of rollDiceForPlayer()");
+  for (let i = 0; i < 2; i += 1) {
```

Comment:
Extra: You could consider making a variable: numberOfPlayers, and then use this variable in your for loop instead of hardcoding 2 here. Reason for this is that in the future, if you would like to increase the number of players, you can simply change the value of the variable instead of finding all the instances where you hardcoded 2 and replacing it.